uPortal-soffit-renderer/src/main/java/org/apereo/portal/soffit/renderer/SoffitRendererController.java

```diff
@@ -249,6 +282,102 @@ private String getCompletePathforParts(final String... parts) {
+
+        final Map<String,Object> rslt = new HashMap<>();
+
+        logger.debug("Processing model attributes for viewName='{}'", viewName);
+
+        for (Map.Entry<AnnotatedElement,Object> y : modelAttributes.entrySet()) {
+            final AnnotatedElement annotatedElement = y.getKey();
+            final Object bean = y.getValue();
+            final SoffitModelAttribute sma = annotatedElement.getAnnotation(SoffitModelAttribute.class);
+            if (attributeAppliesToView(sma, viewName)) {
+                logger.debug("The following  SoffitModelAttribute applies to viewName='{}':  {}", viewName, sma);
+                final String modelAttributeName = sma.value();
+                // Are we looking at a class or a method?
+                if (annotatedElement instanceof Class) {
+                    // The bean itself is the model attribute
+                    rslt.put(modelAttributeName, bean);
+                } else if (annotatedElement instanceof Method) {
+                    Method m = (Method) annotatedElement;
+                    // This Method must NOT have a void return type...
+                    if (m.getReturnType().equals(Void.TYPE)) {
+                        final String msg = "Methods annotated with SoffitModelAttribute must not specify a void return type;  " + m.getName();
+                        throw new IllegalStateException(msg);
+                    }
+                    // Examine the parameters this Method declares and try to match them.
+                    final Class<?>[] parameterTypes = m.getParameterTypes();
+                    final Object[] parameters = new Object[parameterTypes.length];
+                    for (int i=0; i < parameters.length; i++) {
+                        final Class<?> pType = parameterTypes[i];
+                        // At present, these are the parameter types we support...
+                        if (HttpServletRequest.class.equals(pType)) {
+                            parameters[i] = req;
+                        } else if (HttpServletResponse.class.equals(pType)) {
+                            parameters[i] = res;
+                        } else if (PortalRequest.class.equals(pType)) {
+                            parameters[i] = portalRequest;
+                        } else if (Bearer.class.equals(pType)) {
+                            parameters[i] = bearer;
+                        } else if (Preferences.class.equals(pType)) {
+                            parameters[i] = preferences;
+                        } else if (Definition.class.equals(pType)) {
+                            parameters[i] = definition;
+                        } else {
+                            final String msg = "Unsupported parameter type for SoffitModelAttribute method:  " + pType;
+                            throw new UnsupportedOperationException(msg);
+                        }
```

Comment:
Yep -- I agree completely.

I wan to break this method down at the same time I introduce some unit tests.  I think designing the tests will help me choose wisely in breaking it down.

It's probably also a good idea to move this responsibility into another Java class, since it's more than a little bit of logic.